packages/protocol/test/common/proxy.ts

```diff
@@ -222,4 +227,24 @@ contract('Proxy', (accounts: string[]) => {
       })
     })
   })
+
+  it('recovers funds from an incorrectly intialized implementation', async () => {
+    const Freezer: FreezerContract = artifacts.require('Freezer')
+    const GoldToken: GoldTokenContract = artifacts.require('GoldToken')
+    const Registry: RegistryContract = artifacts.require('Registry')
+
+    const freezer = await Freezer.new()
+    const goldToken = await GoldToken.new()
+    const registry = await Registry.new()
+    await registry.setAddressFor(CeloContractName.Freezer, freezer.address)
+    await goldToken.initialize(registry.address)
+
+    await proxy._setImplementation(getSet.address)
+    await goldToken.transfer(proxy.address, 10)
+    const proxyBalance = await web3.eth.getBalance(proxy.address)
+    assert(proxyBalance === '10')
+
+    await recoverFunds(proxy.address, owner)
+    assert((await web3.eth.getBalance(proxy.address)) === '0')
```

Comment:
I think the most important assertion is that the owner actually got the money back :P